doc/source/contribution_and_api/unit_testing.rst

```diff
@@ -0,0 +1,42 @@
+.. _ref_unit_testing_contributing:
+
+Unit Testing
+===============
+
+Unit testing is highly important. It enables to check that the
+changes on the code are efficient and don't have an impact on
+other parts of the code. 
+
+Unit tests are in the ``tests`` directory inside this repository.
+
+Test Validation GitHub workflow
+-------------------------------
```

Comment:
```suggestion
Test Validation GitHub workflow
-------------------------------
```
You should probably mention then the parts of the CICD where the unit tests run.

Furthermore, you could explain the arguments of the ``pytest`` call (maybe too much @RobPasMue ?).

Reply:
Yeah... I wouldn't do that. One thing is explaining what unit tests are and how they are run... Another different thing is copypasting pytest's documentation. I would just redirect to pytest's documentation for further details on its accepted arguments.